### PR TITLE
[FIX] hr_holidays: allow holidays officer to approve leaves

### PR DESCRIPTION
**Pull request by tbe-odoo**
- Holidays officer were not allowed to approve leaves.
- Employee's manager should be able to approve leaves even if it is not
  its leaves manager.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/hr_holidays/models/hr_leave.py [hunk 1]
```diff
@@ -749,17 +749,16 @@ def _check_approval_update(self, state):
                 # use ir.rule based first access check: department, members, ... (see security.xml)
                 holiday.check_access_rule('write')
 
-            if holiday.employee_id == current_employee and not is_manager:
-                raise UserError(_('Only a Leave Manager can approve its own requests.'))
+            if holiday.employee_id == current_employee and not is_officer:
```
Comment (RomainLibert) [inline addons/hr_holidays/models/hr_leave.py]: This was really intended to be limited to managers as validating your own requests is quite powerful.
I don't know if we should allow holiday users to do this

Comment (tbe-odoo) [reply addons/hr_holidays/models/hr_leave.py]: What do you mean by "holiday users"?

Comment (RomainLibert) [reply addons/hr_holidays/models/hr_leave.py]: hr_holidays.group_hr_holidays_user

Comment (tbe-odoo) [reply addons/hr_holidays/models/hr_leave.py]: Yup, that's the xmlid for the Holidays Officer, so it's ok

**Comment by KangOl** [comment]
@robodoo r+

**Pull request closed without merging**